### PR TITLE
Prime 1: Some Phazon Mines logic fixes

### PR DESCRIPTION
**Pull request by XenoWars**
- Removed Boost Ball only method for reverse PMT
- Added Wall Boost in MQA for getting on the spider track to Elevator Access B
- Central Dynamo now accounts for defeating the Drone to go from Save Station Mines B to Dynamo Access
- Added a new node, "Before Fight" which takes all the connections that "Door to Dynamo Access" had
- Added a new node, "Room Bottom" which serves as a central point connecting the three doors
- All doors now connect to "Room Bottom" first before going to other doors, except Dynamo Access to Quarantine Access A
- Added IS method to get the pickup from "Room Bottom" for a situation where you cannot get up to the trigger at all

### CODE REVIEW

File: randovania/games/prime1/json_data/Phazon Mines.txt [hunk 1]
```diff
@@ -1257,7 +1257,8 @@ Extra - aabb: [122.126015, -44.231895, -102.18276, 201.19897, 105.61425, -30.731
   > Door to Quarantine Access A
       All of the following:
           # The Security Drone event triggers door locks on all doors in Central Dynamo except for the door to Quarantine Access A. However, if Samus leaves this room without picking up the item, the doors to Save Station Mines B and Dynamo Access will then permanently be sealed, making it impossible to backtrack through the room.
```
Comment (toasterparty) [inline randovania/games/prime1/json_data/Phazon Mines.txt:1259]: I just now read this comment. Even though it's not your change, I don't think this is true at all. I'm almost certain you can still unlock the doors after passing through once.

File: randovania/games/prime1/json_data/Phazon Mines.txt [hunk 2]
```diff
@@ -1280,15 +1281,29 @@ Extra - aabb: [122.126015, -44.231895, -102.18276, 201.19897, 105.61425, -30.731
...
       All of the following:
-          Morph Ball and Power Bomb and After Invisible Drone
-          Combat (Intermediate) or Normal Damage ≥ 200 or Shoot Ice Beam
+          Morph Ball and Power Bomb
+          Any of the following:
+              All of the following:
+                  After Invisible Drone
+                  Ice Beam or Combat (Intermediate) or Normal Damage ≥ 200
+              Before Invisible Drone and Knowledge (Beginner)
+  > Event - Security Drone
```
Comment (toasterparty) [inline randovania/games/prime1/json_data/Phazon Mines.txt:1298]: Please find a way to add the logic you desire without copy-pasting logic like this. Feel free to add new nodes like `Before Fight` or `Room Center` for example

**Review by toasterparty — COMMENTED** [review]
Please update `CHANGELOG.md` to reflect your changes

**Review by toasterparty — CHANGES_REQUESTED** [review]
see above

File: randovania/games/prime1/json_data/Phazon Mines.txt (hunk 2 above)
Comment (XenoWars) [reply randovania/games/prime1/json_data/Phazon Mines.txt:1298]: I will handle this today

File: randovania/games/prime1/json_data/Phazon Mines.txt [hunk 3]
```diff
@@ -1309,9 +1292,48 @@ Extra - aabb: [122.126015, -44.231895, -102.18276, 201.19897, 105.61425, -30.731
...
+  > Room Bottom
       Trivial
 
+> Before Fight; Heals? False
+  * Layers: default
+  > Door to Dynamo Access
+      After Invisible Drone
+  > Door to Quarantine Access A
+      All of the following:
+          # The Security Drone event triggers door locks on all doors in Central Dynamo except for the door to Quarantine Access A. However, if Samus leaves this room without picking up the item, the doors to Save Station Mines B and Dynamo Access will then permanently be sealed, making it impossible to backtrack through the room.
```
Comment (toasterparty) [inline randovania/games/prime1/json_data/Phazon Mines.txt:1304]: This is redundant with the `Room Bottom -> Door to Quarantine Access A` logic. The player should just be able to go `Door to Dynamo Access -> Room Bottom (Trivial) -> Door to Quarantine Access A`, thus avoiding duplicate logic.

**Review by toasterparty — CHANGES_REQUESTED** [review]
The IS on [line 1288](https://github.com/randovania/randovania/blob/3ea01e754589a2ef05bec1b5e4cfc0c2e32fb17a/randovania/games/prime1/json_data/Phazon%20Mines.txt#L1288) is more appropriate from the `Before Fight` node since you don't actually have to kill the drone, but you do take damage from ISing it.

File: randovania/games/prime1/json_data/Phazon Mines.txt (hunk 1 above)
Comment (XenoWars) [reply randovania/games/prime1/json_data/Phazon Mines.txt]: I now understand what you mean. Yeah. the doors do still unlock after the drone is defeated and the item collected.

File: randovania/games/prime1/json_data/Phazon Mines.txt [hunk 4]
```diff
@@ -1309,9 +1292,48 @@ Extra - aabb: [122.126015, -44.231895, -102.18276, 201.19897, 105.61425, -30.731
...
+          Any of the following:
+              Knowledge (Beginner)
+              All of the following:
+                  After Invisible Drone
+                  Ice Beam or Combat (Intermediate) or Normal Damage ≥ 200
+  > Door to Save Station Mines B
+      Before Invisible Drone or Combat (Intermediate) or Normal Damage ≥ 200 or Shoot Ice Beam
+  > Pickup (Main Power Bombs)
+      All of the following:
+          # Achieving the Infinite Speed without going up to hit the fight trigger: https://youtu.be/LRmJn2BAYwc
```
[the remark this answers is not in the document]
Comment (XenoWars) [reply randovania/games/prime1/json_data/Phazon Mines.txt:1326]: That I don't know but seeing how the fences aren't up and the drone isnt active, I would think so. I might have to put "Before Invisible Drone"

[the remark this answers is not in the document]
Comment (toasterparty) [reply randovania/games/prime1/json_data/Phazon Mines.txt:1326]: There's the possibility of bumping into the firing squad in the next room and still taking 100+ E of contact damage. Better to either test these things or go on the safe side of guessing

**Comment by XenoWars** [comment]
I had someone test the damages for doing the IS coming from Quarantine Access A

**Review by toasterparty — APPROVED** [review]

**Pull request merged**